src/main/java/com/code_intelligence/jazzer/junit/FuzzerDictionary.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2023 Code Intelligence GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.code_intelligence.jazzer.junit;
+
+import com.code_intelligence.jazzer.utils.Log;
+import java.io.*;
+import java.lang.annotation.*;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.platform.commons.util.ClassLoaderUtils;
+
+/**
+ * Class that manages dictionaries for fuzz tests. The {@link WithDictionary} and {@link
+ * WithDictionaryFile} annotations are added to {@link FuzzTest}s to indicate that these
+ * dictionaries should be used for fuzzing this function. All tokens from all the sources will be
+ * added into a single merged dictionary file as libfuzzer can only accept a single {@code -dict}
+ * flag.
+ *
+ * <p>Syntax for dictionaries can be found <a
+ * href="https://llvm.org/docs/LibFuzzer.html#dictionaries">here</a>.
+ */
+public class FuzzerDictionary {
+  private static final String DICTIONARY_PREFIX = "jazzer-";
+  private static final String DICTIONARY_SUFFIX = ".dict";
+
+  /**
+   * Create a temporary dictionary file for use during a fuzzing run based on the tokens found
+   * within {@code inline} and {@code files}.
+   *
+   * @param inline List of {@link WithDictionary} annotations that directly hold static token values
+   *     to use in the dictionary
+   * @param files List of {@link WithDictionaryFile} annotations that reference dictionary files to
+   *     include
+   * @return Optional containing the path to the created file, or nothing if {@code inline} and
+   *     {@code files} are both empty
+   * @throws IOException
+   */
+  public static Optional<String> createDictionaryFile(
+      List<WithDictionary> inline, List<WithDictionaryFile> files) throws IOException {
+    int sources = inline.size() + files.size();
+    if (sources == 0) {
+      return Optional.empty();
+    }
+
+    Stream<String> joined = Stream.concat(getInlineTokens(inline), getFileTokens(files));
+
+    File f = File.createTempFile(DICTIONARY_PREFIX, DICTIONARY_SUFFIX);
```

Comment:
`Files.createTempFile` is preferred as it builds on the more modern `Path` class and also uses more restrictive permissions. `path.toFile().deleteOnExit()` still works.